code/game/machinery/iv_drip.dm

```diff
@@ -1,224 +1,684 @@
+		usr.visible_message("<b>[usr]</b> knocks \the [src] down!", "You knock \the [src] down!")
+		do_crash()
+		return
+	var/list/options = list(
+		"Transfer Rate" = image('icons/mob/screen/radial.dmi', "radial_transrate"),
+		"Toggle Mode" = image('icons/mob/screen/radial.dmi', "iv_mode"),
+		"Toggle Stop" = image('icons/mob/screen/radial.dmi', "iv_stop"),
+		"Toggle Valve" = image('icons/mob/screen/radial.dmi', "iv_valve"),
+		"Toggle EPP" = image('icons/mob/screen/radial.dmi', "iv_epp"))
+	var/chosen_action = show_radial_menu(usr, src, options, require_near = TRUE, radius = 42, tooltips = TRUE)
+	if(!chosen_action)
+		return
+	switch(chosen_action)
+		if("Transfer Rate")
+			transfer_rate()
+		if("Toggle Mode")
+			toggle_mode()
+		if("Toggle Stop")
+			toggle_stop()
+		if("Toggle Valve")
+			toggle_valve()
+		if("Toggle EPP")
+			toggle_epp()
+		else
+			return
 
+/obj/machinery/iv_drip/proc/do_crash()
+	cut_overlays()
+	visible_message(SPAN_WARNING("\The [src] falls over with a buzz, spilling out it's contents!"))
+	flick("iv_crash[is_loose ? "" : "_tank_[tank_type]"]", src)
+	playsound(src, 'sound/items/drop/prosthetic.ogg', 50)
+	spill()
+	tipped = TRUE
+	icon_state = "iv_stand_tipped"
+	update_icon()
+
+/obj/machinery/iv_drip/proc/spill()
+	var/turf/dropspot = get_turf(src)
+	if(breath_mask)
+		if(!breather)
+			breath_mask.forceMove(dropspot)
+			breath_mask.tumble(rand(1,3))
+			breath_mask.SpinAnimation(4, 2)
+			breath_mask = null
+		else
+			src.visible_message(SPAN_WARNING("\The [breath_mask] pulls \the [breather] down with \the [src]!"), SPAN_WARNING("\The [breath_mask] pulls you down with \the [src]!"))
+			breather.forceMove(dropspot)
+			breather.Weaken(4)
+	if(beaker)
+		beaker.forceMove(dropspot)
+		beaker.tumble(rand(1,3))
+		beaker.SpinAnimation(4, 2)
+		beaker = null
+	if(attached)
+		iv_rip()
+	if(tank)
+		if(is_loose)
+			tank.forceMove(dropspot)
+			tank.tumble(rand(1,2))
+			tank.SpinAnimation(4, 2)
+			tank = null
+			tank_type = null
+			if(breather)
+				if(breather.internals)
+					breather.internals.icon_state = "internal0"
+				breather.internal = null
+				valve_open = FALSE
+				tank_active = FALSE
+				epp_active = FALSE
+		src.visible_message("\The [tank] rattles, but remains firmly secured to \the [src].")
+
+/obj/machinery/iv_drip/proc/iv_rip()
+	attached.visible_message(SPAN_WARNING("The needle is ripped out of [attached]'s [vein.name]."), SPAN_DANGER("The needle <B>painfully</B> rips out of your [vein.name]."))
+	vein.take_damage(brute = 5, damage_flags = DAM_SHARP)
+	vein = null
+	attached = null
+
+/obj/machinery/iv_drip/proc/tank_off()
+	tank.forceMove(src)
+	if(breather.internals)
+		breather.internals.icon_state = "internal0"
+	breather.internal = null
+	tank_active = FALSE
+	valve_open = FALSE
+	epp_active = FALSE
+
+/obj/machinery/iv_drip/proc/toggle_check()
+	if(!ishuman(usr) && !issilicon(usr))
+		to_chat(usr, SPAN_WARNING("This mob cannot operate the controls!"))
+		return
+	if(!isliving(usr))
```

Comment:
If we're not living, then we already failed the check on L558 because human and silicon are both `/mob/living` subtypes.